crates/noirc_evaluator/src/ssa/acir_gen/operations/store.rs

```diff
@@ -19,41 +24,47 @@ pub(crate) fn evaluate(
 ) -> Option<InternalVar> {
     if let Operation::Store { array_id, index, value, predicate, .. } = *store {
         //maps the address to the rhs if address is known at compile time
-        let index = var_cache.get_or_compute_internal_var_unwrap(index, evaluator, ctx);
-        let value = var_cache.get_or_compute_internal_var_unwrap(value, evaluator, ctx);
+        let index_var = var_cache.get_or_compute_internal_var_unwrap(index, evaluator, ctx);
+        let value_var = var_cache.get_or_compute_internal_var_unwrap(value, evaluator, ctx);
+        let value_with_predicate = if let Some(predicate) = predicate {
+            if predicate.is_dummy() || ctx.is_one(predicate) {
+                value_var
+            } else if ctx.is_zero(predicate) {
+                return None;
+            } else {
+                let pred = var_cache.get_or_compute_internal_var_unwrap(predicate, evaluator, ctx);
+                let dummy_load =
+                    load::evaluate(array_id, index, acir_mem, var_cache, None, evaluator, ctx)
+                        .unwrap();
```

Comment:
`unwrap` to expect with reason

Reply:
Done, I return an error instead of unwrapping.